awscli/customizations/s3/subcommands.py

```diff
@@ -750,12 +750,35 @@ class SyncCommand(S3TransferCommand):
                 [METADATA, METADATA_DIRECTIVE]
 
 
-class MbCommand(S3TransferCommand):
+class MbCommand(S3Command):
     NAME = 'mb'
     DESCRIPTION = "Creates an S3 bucket."
     USAGE = "<S3Uri>"
-    ARG_TABLE = [{'name': 'paths', 'nargs': 1, 'positional_arg': True,
-                  'synopsis': USAGE}]
+    ARG_TABLE = [{'name': 'path', 'positional_arg': True, 'synopsis': USAGE}]
+
+    def _run_main(self, parsed_args, parsed_globals):
+        super(MbCommand, self)._run_main(parsed_args, parsed_globals)
+
+        path = parsed_args.path
+        if path.startswith('s3://'):
+            path = path[5:]
+        bucket, _ = find_bucket_key(path)
+
+        bucket_config = {'LocationConstraint': self.client.meta.region_name}
+        params = {'Bucket': bucket}
+        if self.client.meta.region_name != 'us-east-1':
+            params['CreateBucketConfiguration'] = bucket_config
+
+        try:
```

Comment:
@jamesls this is the relevant diff/commit: https://github.com/aws/aws-cli/pull/2161/commits/4d99056fc8ea6500abf59e413e53945e4f4be613. So I do agree that the result processor, especially since a thread is spun up, is not really needed for this command. I think the main part of concern for me was that before with that commit we had two different places where we did the result printing logic and I was wondering if there was a way we could expose even just a helper function that consolidates this logic and have both the ``ResultPrinter`` and ``MbCommand`` can use it as if we need to update the printing logic ever, there is a good chance that we forget to update both places if we do not consolidate. Does that make sense?